Locker/Locker/Locker.swift

```diff
@@ -0,0 +1,246 @@
+//
+//  Locker.swift
+//  Locker
+//
+//  Created by Zvonimir Medak on 19.10.2021..
+//  Copyright © 2021 Infinum. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+@objcMembers
+public class Locker: NSObject {
+
+    // MARK: - Public properties
+
+    public static var userDefaults: UserDefaults? {
+        get {
+            return currentUserDefaults == nil ? UserDefaults.standard : currentUserDefaults
+        }
+        set {
+            currentUserDefaults = currentUserDefaults == nil ? UserDefaults.standard : newValue
+        }
+    }
+
+    public static var biometricsSettingsDidChange: Bool {
+        LockerHelpers.biometricsSettingsChanged
+    }
+
+    public static var isRunningFromTheSimulator: Bool {
+    #if targetEnvironment(simulator)
+        return true
+    #else
+        return false
+    #endif
+    }
+
+    public static var deviceSupportsAuthenticationWithBiometrics: BiometricsType {
+        LockerHelpers.deviceSupportsAuthenticationWithBiometrics
+    }
+
+    public static var configuredBiometricsAuthentication: BiometricsType {
+        LockerHelpers.configuredBiometricsAuthentication
+    }
+
+    // MARK: - Private properties
+
+    private static var currentUserDefaults: UserDefaults?
+
+    // MARK: - Handle secrets (store, delete, fetch)
+
+    @objc(setSecret:forUniqueIdentifier:error:)
+    public static func setSecret(_ secret: String, for uniqueIdentifier: String) throws {
+    #if targetEnvironment(simulator)
+        Locker.userDefaults?.set(secret, forKey: uniqueIdentifier)
+    #else
+        setSecretForDevice(secret, for: uniqueIdentifier) { error in
+            guard let error = error else {
+                return
+            }
+            throw error
+        }
```

Comment:
```suggestion
        setSecretForDevice(secret, for: uniqueIdentifier, completion: { error in
            guard let error = error else {
                return
            }
            throw error
        })
```
Let's avoid trailing closures as in that way, we are losing the info on what's going on. So, without this, we don't know if this will be called only on error or it will be called on success  as well (completion in general)? :)